### PR TITLE
Update jest to v26 along with packages related to it

### PR DESCRIPTION
**Pull request by abnud1**
## Summary
The only problem is jest-snapshot-serializer-raw, it hasn't been updated for 2 years but tests fail without it and I couldn't find a replacement.

I see only two options here:

1. migrate the snapshot tests to another serializer or to jest default serializer.
2. fork jest-snapshot-serializer-raw

## Test Plan
run the tests, tests that passed before should not fail after the upgrade, and tests that failed before should fail too

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,5 +127,13 @@
...
     "merge-fork": "node ./scripts/merge-fork/merge-fork.js",
     "replace-fork": "node ./scripts/merge-fork/replace-fork.js"
+  },
+  "dependencies": {
+    "eslint-plugin-jest": "^24.0.2",
+    "jest": "^26.4.2",
+    "jest-cli": "^26.4.2",
+    "jest-diff": "^26.4.2",
+    "jest-environment-jsdom": "^26.3.0",
+    "jest-snapshot-serializer-raw": "^1.1.0"
```
Comment (bvaughn) [inline package.json:137]: These should be `devDependencies`.

Comment (abnud1) [reply package.json:137]: Yes, Sorry for that my mistake

**Comment by gaearon** [comment]
Looks like there's CI failures. Ideas?

**Comment by bvaughn** [comment]
[This one](https://app.circleci.com/pipelines/github/facebook/react/5334/workflows/094d0a36-af3f-460b-9d56-aa650a919f7a/jobs/214285) is a lint failure.

> TypeError: jest.requireActual is not a function

This one looks to be directly caused by one of the changed files.

**Comment by abnud1** [comment]
@gaearon the failure happens because fixtures was still using jest 25 in which `require.requireActual` must be used not `jest.requireActual`

**Comment by abnud1** [comment]
@bvaughn the lint failure comes from updating eslint-plugin-jest

**Comment by abnud1** [comment]
@gaearon Ok it seems that fixtures/dom is still using Jest 25 even after updating the jest version in all package.json files I could find. maybe because of `--frozen-lock`?

Anyway even after changing `jest.requireActual` with `require.requireActual` I get this error now:

> TypeError: Cannot read property 'unstable_now' of undefined

This problem is probably because mocking `scheduler` module didn't work for some reason

File: scripts/jest/config.base.js [hunk 2]
```diff
@@ -25,5 +25,5 @@ module.exports = {
   testSequencer: require.resolve('./jestSequencer'),
 
   // TODO: Upgrade to Jest 26 which uses jsdom 16 by default.
```
Comment (SimenB) [inline scripts/jest/config.base.js:27]: remove this line?

Comment (abnud1) [reply scripts/jest/config.base.js:27]: unfortunately that didn't fix the problem

Comment (SimenB) [reply scripts/jest/config.base.js:27]: I just meant removing the todo comment

File: package.json [hunk 3]
```diff
@@ -134,5 +133,7 @@
     "merge-fork": "node ./scripts/merge-fork/merge-fork.js",
     "replace-fork": "node ./scripts/merge-fork/replace-fork.js"
   },
-  "dependencies": {}
-}
\ No newline at end of file
+  "dependencies": {
+    "babel-eslint": "^10.1.0"
```
Comment (bvaughn) [inline package.json:137]: Not a dependency.

**Review by bvaughn — COMMENTED** [review]
Initially a little skeptical about why bumping dependencies resulted in 25,000+ lines added to the yarn lockfiles.

**Comment by matthargett** [comment]
> Initially a little skeptical about why bumping dependencies resulted in 25,000+ lines added to the yarn lockfiles.

FWIW, the jest repo's yarn.lock has increased more than 10% in the last year. it's over 680kb now. I don't think that by itself should indicate concern or a problem. Since it doesn't have a resulting bundle size impact, would timing a fully-cached 'yarn install' before and after could be enough to quell your skepticism?

**Pull request closed without merging**